src/cmd/ksh93/sh/io.c

```diff
@@ -628,14 +616,12 @@ int sh_open(const char *path, int flags, ...) {
             fd = 2;
         }
 
-#ifdef O_SERVICE
         if (fd < 0) {
             if ((fd = inetopen(path + 5, flags, onintr, shp)) < 0 && errno != ENOTDIR) return -1;
```

Comment:
This looks like specific to `mkservice` builtin.

Reply:
Yes it does look like that and it's wrong :smile: The `inetopen()` function is what implements support for pathnames as in this example: `echo hello >/dev/tcp/localhost/1234`. Personally I would just remove it as I don't think ksh is the right tool for network communication. But that would break backward compatibility with ksh93u+. The only part of this change that is specific to `mkservice` is the `int server` var and the code predicated on it being true. That I did remove since `server` can never be true now that we've removed `mkservice`.